src/dom/rect.js

```diff
@@ -54,11 +54,39 @@ export default class Rect {
 			enumerable: false
 		} );
 
-		if ( isElement( source ) || isRange( source ) ) {
-			source = source.getBoundingClientRect();
-		}
+		// Acquires all the rect properties from the passed source.
+		//
+		// @private
+		// @param {ClientRect|module:utils/dom/rect~Rect|Object} source}
+		const setProperties = source => {
+			rectProperties.forEach( p => this[ p ] = source[ p ] );
+		};
 
-		rectProperties.forEach( p => this[ p ] = source[ p ] );
+		if ( isElement( source ) ) {
+			setProperties( source.getBoundingClientRect() );
```

Comment:
I don't like how many times `setProperties` is executed in this file. Couldn't the rect be assigned to some value and later on the values be set once?